rodi/__init__.py

```diff
@@ -21,8 +21,9 @@
 )
 
 if sys.version_info >= (3, 9):  # pragma: no cover
-    # Python 3.9
-    from typing import _no_init_or_replace_init
+    from typing import _no_init_or_replace_init as _no_init
+elif sys.version_info >= (3, 8):  # pragma: no cover
+    from typing import _no_init

```

Comment:
So, `Protocol` was introduced in Python 3.8, but since then it has changed the name of the
function it uses as a default `__init__` .

It was `_no_init` in 3.8, but that changed quickly to `_no_init_or_replace_init` for >=3.9